store/tikv/oracle/oracles/pd_test.go

```diff
@@ -37,3 +38,43 @@ func TestPDOracle_UntilExpired(t *testing.T) {
 		t.Errorf("waitTs shoulb be %d but got %d", int64(lockAfter+lockExp), waitTs)
 	}
 }
+
+func TestPdOracle_GetStaleTimestamp(t *testing.T) {
+	o := oracles.NewEmptyPDOracle()
+	start := time.Now()
+	oracles.SetEmptyPDOracleLastTs(o, oracle.ComposeTS(oracle.GetPhysical(start), 0))
+	oracles.SetEmptyPDOracleLastArrivalTs(o, oracle.ComposeTS(oracle.GetPhysical(start), 0))
+	ts, err := o.GetStaleTimestamp(context.Background(), 10)
+	if err != nil {
+		t.Errorf("%v\n", err)
+	}
+
+	duration := start.Sub(oracle.GetTimeFromTS(ts))
+	if duration > 12*time.Second || duration < 8*time.Second {
+		t.Errorf("stable TS have accuracy err, expect: %d +-2, obtain: %d", 10, duration)
+	}
+
+	_, err = o.GetStaleTimestamp(context.Background(), 1e12)
+	if err == nil {
+		t.Errorf("expect exceed err but get nil")
+	}
+
+	_, err = o.GetStaleTimestamp(context.Background(), -2)
+	if err != nil {
+		t.Errorf("%v\n", err)
+	}
```

Comment:
Should assert error here.

Reply:
Thanks for your review, I agree with you.